Modules/Loadable/Texts/Widgets/qMRMLTextWidget.h

```diff
@@ -0,0 +1,131 @@
+/*==============================================================================
+
+  Copyright (c) Laboratory for Percutaneous Surgery (PerkLab)
+  Queen's University, Kingston, ON, Canada. All Rights Reserved.
+
+  See COPYRIGHT.txt
+  or http://www.slicer.org/copyright/copyright.txt for details.
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  This file was originally developed by Kyle Sunderland, PerkLab, Queen's University
+  and was supported through CANARIE's Research Software Program, and Cancer
+  Care Ontario.
+
+==============================================================================*/
+
+#ifndef __qMRMLTextWidget_h
+#define __qMRMLTextWidget_h
+
+// SlicerQt includes
+#include "qSlicerWidget.h"
+
+// Text widgets includes
+#include "qSlicerTextsModuleWidgetsExport.h"
+#include "ui_qMRMLTextWidget.h"
+
+class vtkMRMLNode;
+class vtkMRMLTextNode;
+class qMRMLTextWidgetPrivate;
+
+/// \ingroup Slicer_QtModules_Texts
+class Q_SLICER_MODULE_TEXTS_WIDGETS_EXPORT qMRMLTextWidget : public qSlicerWidget
+{
+  Q_OBJECT
+
+public:
+  typedef qSlicerWidget Superclass;
+  qMRMLTextWidget(QWidget *parent=nullptr);
+  ~qMRMLTextWidget() override;
+
+  Q_PROPERTY(bool readOnly READ readOnly WRITE setReadOnly)
+  Q_PROPERTY(bool continuousUpdate READ continuousUpdate WRITE setContinuousUpdate)
+
+  /// Get the table node the columns are edited of.
+  Q_INVOKABLE vtkMRMLTextNode* mrmlTextNode() const;
+  Q_INVOKABLE vtkMRMLNode* mrmlNode() const;
+
+  /// Returns true if the text editor is read only
+  bool readOnly();
+
+  /// Returns true if the text editor is continuously updated from the text node and vice versa
+  bool continuousUpdate();
```

Comment:
What do you think of using `autoUpdate` ?  This would be consistent with the  `updateWidget` and `updateFinished` method names

Reply:
Sounds good! autoUpdate() is much better than continuousUpdate().

Reply:
Done!